core/src/main/java/org/jdbi/v3/core/Jdbi.java

```diff
@@ -405,10 +416,10 @@ public <R, X extends Exception> R withHandle(HandleCallback<R, X> callback) thro
             sqlStatements.setAttachAllStatementsForCleanup(sqlStatements.isAttachCallbackStatementsForCleanup());
 
             handleSupplier = ConstantHandleSupplier.of(h);
-            threadHandleSupplier.set(handleSupplier);
+            getHandleSupplierHolder().setHandleSupplier(handleSupplier);
             return decoratedCallback.withHandle(h);
         } finally {
-            threadHandleSupplier.remove();
+            getHandleSupplierHolder().clearHandleSupplier();
```

Comment:
should you look up the handle supplier holder here again? seems like the lookup can be done once, and you always set / clear on the same instance.

Reply:
can we move the review of the handle supplier holder code to #2523 ?